app/views/frontendapi.py

```diff
@@ -81,3 +82,104 @@ def _result_to_dict(self, person):
             # TODO(nworden): ask Travis/Pete if we should do something about
             # external photos here.
         }
+
+
+class CreateView(FrontendApiBaseView):
+    """View for creating a new record."""
+
+    def setup(self, request, *args, **kwargs):
+        super(CreateView, self).setup(request, *args, **kwargs)
+        self.params.read_values(
+            post_params={
+                'add_note': utils.validate_yes,
+                'age': utils.validate_age,
+                'alternate_family_names': utils.strip,
+                'alternate_given_names': utils.strip,
+                'author_email': utils.strip,
+                'author_made_contact': utils.validate_yes,
+                'author_name': utils.strip,
+                'author_phone': utils.strip,
+                'users_own_email': utils.strip,
+                'users_own_phone': utils.strip,
+                'clone': utils.validate_yes,
+                'description': utils.strip,
+                'email_of_found_person': utils.strip,
+                'family_name': utils.strip,
+                'given_name': utils.strip,
+                'home_city': utils.strip,
+                'home_country': utils.strip,
+                'home_neighborhood': utils.strip,
+                'home_postal_code': utils.strip,
+                'home_state': utils.strip,
+                'last_known_location': utils.strip,
+                'note_photo': utils.validate_image,
+                'note_photo_url': utils.strip,
+                'phone_of_found_person': utils.strip,
+                'photo': utils.validate_image,
+                'photo_url': utils.strip,
+                'referrer': utils.strip,
+                'sex': utils.validate_sex,
+                'source_date': utils.strip,
+                'source_name': utils.strip,
+                'source_url': utils.strip,
+                'status': utils.validate_status,
+                'text': utils.strip,
+                'own_info': utils.validate_yes,
+            })
+        if request.method == 'POST':
+            profile_urls = []
+            profile_field_index = 0
+            while True:
```

Comment:
this infinite loop makes me nervous. Could you not just loop over the keys in self.request.POST instead, and take the ones that match `profile-url-[\d]+`?

Reply:
I'd prefer not to do that, because a) it would make it inconvenient to maintain their order and b) it'd be marginally less efficient. Instead I changed it to a for loop over range(size of POST dict) (since we certainly don't need to check values higher than that).

Reply:
JK, request.POST doesn't have a size function, and you can't get the size without having it iterate through everything afaict, so I just hard-coded a limit.